GoogleSignIn/Tests/Unit/GIDAuthStateMigrationTest.m

```diff
@@ -113,44 +122,49 @@ - (void)tearDown {
-                                        callbackPath:kCallbackPath
-                                        keychainName:kKeychainName
-                                      isFreshInstall:NO];
+  [[_mockGTMKeychainStore expect] saveAuthSession:OCMOCK_ANY error:OCMArg.anyObjectRef];
+
+  [self setUpCommonExtractAuthorizationMocksWithFingerPrint:kSavedFingerprint];
+
+  GIDAuthStateMigration *migration =
+      [[GIDAuthStateMigration alloc] initWithKeychainStore:_mockGTMKeychainStore];
+  [migration migrateIfNeededWithTokenURL:[NSURL URLWithString:kTokenURL]
+                            callbackPath:kCallbackPath
+                            keychainName:kKeychainName
+                          isFreshInstall:NO];
 }
 
 - (void)testMigrateIfNeeded_HasPreviousMigration {
   [[[_mockUserDefaults stub] andReturn:_mockUserDefaults] standardUserDefaults];
-  [[[_mockUserDefaults expect] andReturnValue:@YES]
-      boolForKey:kMigrationCheckPerformedKey];
-
-  [GIDAuthStateMigration migrateIfNeededWithTokenURL:[NSURL URLWithString:kTokenURL]
-                                        callbackPath:kCallbackPath
-                                        keychainName:kKeychainName
-                                      isFreshInstall:NO];
+  [[[_mockUserDefaults expect] andReturnValue:@YES] boolForKey:kMigrationCheckPerformedKey];
+  [[_mockUserDefaults reject] setBool:YES forKey:kMigrationCheckPerformedKey];
+
+  GIDAuthStateMigration *migration =
+      [[GIDAuthStateMigration alloc] initWithKeychainStore:_mockGTMKeychainStore];
+  [migration migrateIfNeededWithTokenURL:[NSURL URLWithString:kTokenURL]
+                            callbackPath:kCallbackPath
+                            keychainName:kKeychainName
+                          isFreshInstall:NO];
 }
 
 - (void)testMigrateIfNeeded_KeychainFailure {
   [[[_mockUserDefaults stub] andReturn:_mockUserDefaults] standardUserDefaults];
-  [[[_mockUserDefaults expect] andReturnValue:@NO]
-      boolForKey:kMigrationCheckPerformedKey];
-  [[[_mockGIDAuthStateMigration expect] andReturn:_mockGTMAppAuthFetcherAuthorization]
-      extractAuthorizationWithTokenURL:[NSURL URLWithString:kTokenURL] callbackPath:kCallbackPath];
-  [[[_mockGTMAppAuthFetcherAuthorization expect] andReturnValue:[NSNumber numberWithBool:NO]]
-      saveAuthorization:_mockGTMAppAuthFetcherAuthorization toKeychainForName:kKeychainName];
-
-  [GIDAuthStateMigration migrateIfNeededWithTokenURL:[NSURL URLWithString:kTokenURL]
-                                        callbackPath:kCallbackPath
-                                        keychainName:kKeychainName
-                                      isFreshInstall:NO];
+  [[[_mockUserDefaults expect] andReturnValue:@NO] boolForKey:kMigrationCheckPerformedKey];
+
+  NSError *keychainSaveError = [NSError new];
+  OCMStub([_mockGTMKeychainStore saveAuthSession:OCMOCK_ANY error:[OCMArg setTo:keychainSaveError]]);
```

Comment:
Just curious on the plan for OCMock v3 or v2 style (https://ocmock.org/ocmock3/). Historically all tests in this repo are in OCMock v2 style, e.g., `[[mock stub] method];`. Do we want to start to have new code in OCMock v3 style, e.g., `OCMStub([mock method]);`, and potentially towards the goal of convert all tests to the new style, or shall we keep it consistent with OCMock v2 style?

Reply:
I think the v3 style is better. Ultimately, our plan is to reduce and eventually remove mocking. Until then, we can gradually move over to the v3 style.